spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/src/main/java/org/springframework/cloud/dataflow/module/deployer/kubernetes/ContainerFactory.java

```diff
@@ -0,0 +1,16 @@
+package org.springframework.cloud.dataflow.module.deployer.kubernetes;
+
+import org.springframework.cloud.dataflow.core.ModuleDeploymentRequest;
+
+import io.fabric8.kubernetes.api.model.Container;
+
+/**
+ * Defines how a Kubernetes {@link Container} is created. 
+ * 
+ * @author Florian Rosenberg
+ */
+public interface ContainerFactory {
```

Comment:
I think it would be better if we kept this PR "simple" (it's already quite big and will become more complicated when trying to uncouple redis).

The concept of `ContainerFactory` should be added as a separate PR, later (if ever -- that is another story)